src/Peachpie.CodeAnalysis/CodeGen/VariableReference.cs

```diff
@@ -1176,8 +1181,11 @@ public override void EmitInit(CodeGenerator cg)
 
             if (source == target)
             {
-                // 2a. (source == target): Pass (inplace copy)
-                source.EmitPass(cg);
+                // 2a. (source == target): Pass (inplace copy and dereference) if necessary
+                if (!SkipPass)
```

Comment:
do not check it in here, add the condition inside `EmitPass` ...

Reply:
Updated in 96eb30660bd40b4ca997494b0068443601f14761.